tests/components/cloudflare/test_config_flow.py

```diff
@@ -0,0 +1,166 @@
+
+    assert result["result"]
+    assert result["result"].unique_id == USER_INPUT_ZONE[CONF_ZONE]
+
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_user_form_cannot_connect(hass, cfupdate_flow):
+    """Test we handle cannot connect error."""
+    instance = cfupdate_flow.return_value
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER}
+    )
+
+    instance.get_zones.side_effect = CloudflareConnectionException()
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        USER_INPUT,
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_user_form_invalid_auth(hass, cfupdate_flow):
+    """Test we handle invalid auth error."""
+    instance = cfupdate_flow.return_value
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER}
+    )
+
+    instance.get_zones.side_effect = CloudflareAuthenticationException()
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        USER_INPUT,
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "invalid_auth"}
+
+
+async def test_user_form_invalid_zone(hass, cfupdate_flow):
+    """Test we handle invalid zone error."""
+    instance = cfupdate_flow.return_value
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER}
+    )
+
+    instance.get_zones.side_effect = CloudflareZoneException()
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        USER_INPUT,
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "invalid_zone"}
+
+
+async def test_user_form_unexpected_exception(hass, cfupdate_flow):
+    """Test we handle unexpected exception."""
+    instance = cfupdate_flow.return_value
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER}
+    )
+
+    instance.get_zones.side_effect = Exception()
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        USER_INPUT,
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "unknown"}
+
+
+async def test_user_form_single_instance_allowed(hass):
+    """Test that configuring more than one instance is rejected."""
+    entry = MockConfigEntry(domain=DOMAIN, data=ENTRY_CONFIG)
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={CONF_SOURCE: SOURCE_USER},
+        data=USER_INPUT,
+    )
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "single_instance_allowed"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)